src/nxdoc/web-ui/web-ui-tutorials/how-to-use-rich-text-editor-rte-for-article-publishing.md

```diff
@@ -0,0 +1,155 @@
+
+In this page we will give you an overview of how you can use the rich text editor and configure it to your liking. You will also find some ideas to configure or extend it further.
+
+{{#> callout type='info'}}
+This how-to requires Nuxeo LTS 2021 or newer.
+{{/callout}}
+
+## Using the Rich Text Editor
+
+To test the default behavior of the rich text editor
+
+1. In a Workspace or a Folder, create one or several `Picture` documents.
+1. In a Workspace or a Folder, create a new `Note` document and keep the `HTML` format.
+1. Type your content and save it directly while being in the document view. Use the rich text editor toolbar to leverage the various options.
+
+To reference one or several pictures stored in Nuxeo
+
+1. Click on the `insert images from existing documents` [//TODO insert icon] toolbar icon
+1. Search for your content using the quick filter (fulltext search) option, select as many images as you want and press the `Select` button.
+
+The search form, the query used to filter pictures and the results presentation are all configurable using Nuxeo Studio.
+
+## Accessing the Content of the Rich Text Editor
+
+As usual with Nuxeo, anything in the UI can be accessed through an external system by using the REST API. //TODO link to the REST API doc
+
+When retrieving a Note document, the content will be referenced in the corresponding document property, e.g.:
+
+```
+[...]
+"properties": {
+  "note:note": "Some text and now an image <img src=\&quot;https://nuxeo-server-url.nuxeo.com/nuxeo/nxfile/default/793dc397-2ecc-46ee-8840-92d1d88faed6/file:content/Away.jpg\&quot;>"
+[...]
+```
+
+Any image you referenced is stored in a way that lets you parse the content and adapt it to your needs. Your own tool can easily adapt that URL to call further Nuxeo services, for example to request a particular rendition.
+
+## Configuring the Query
+
+By default, inserting images in the rich text editor will reference the current content stored in any document having the `Picture` facet.
+
+Depending on your needs, you may want to change the query to retrieve specific document types, documents in a specific state (e.g. only return approved images). You may even want users to pick only the latest version available for greater control.
+
+Here's a quick example to retrieve only the latest version of an image.
+
+//TODO check with UI team what we should provide by default in the template: e.g. results layout for people to have the custom code or not to avoid them having to maintain it?
+
+1. In Studio Modeler, go to **External Templates**; import the **Rich Text Editor Default Configuration** template.
+1. Go to **Configuration** > **Page Providers**; select the `document_picker` page provider.
+1. Replace the query filter with the following:
+
+```
+ecm:mixinType != 'HiddenInNavigation' AND ecm:mixinType = 'Picture' AND ecm:isTrashed = 0 AND ecm:isVersion = 1 AND ecm:isLatestVersion = 1
```

Comment:
FYI, this removes the `file:content IS NOT NULL` query part, which would prevent you from getting `Picture` documents without images.

Reply:
Ah, I'll add it back, thanks Miguel!

Reply:
Change incorporated.